src/rpc/misc.cpp

```diff
@@ -276,31 +276,19 @@ static UniValue verifymessage(const JSONRPCRequest& request)
     std::string strSign     = request.params[1].get_str();
     std::string strMessage  = request.params[2].get_str();
 
-    CTxDestination destination = DecodeDestination(strAddress);
-    if (!IsValidDestination(destination)) {
-        throw JSONRPCError(RPC_TYPE_ERROR, "Invalid address");
+    switch (MessageVerify(strAddress, strSign, strMessage)) {
+        case MessageVerificationResult::ERR_INVALID_ADDRESS:
+            throw JSONRPCError(RPC_TYPE_ERROR, "Invalid address");
+        case MessageVerificationResult::ERR_ADDRESS_NO_KEY:
+            throw JSONRPCError(RPC_TYPE_ERROR, "Address does not refer to key");
+        case MessageVerificationResult::ERR_MALFORMED_SIGNATURE:
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Malformed base64 encoding");
+        case MessageVerificationResult::ERR_PUBKEY_NOT_RECOVERED:
+        case MessageVerificationResult::ERR_NOT_SIGNED:
+            return false;
+        case MessageVerificationResult::OK:
+            return true;
     }
```

Comment:
You need another `return false` statement after this `switch`.

Reply:
Nice catch. I'm not getting that `-Wreturn-type` warning on macOS. Only one of the Travis machines noticed it: https://travis-ci.org/bitcoin/bitcoin/jobs/649943882?utm_medium=notification&utm_source=github_status

Reply:
Added `return false;`, thanks!